butler-bot/bot.js

```diff
@@ -63,6 +73,38 @@ client.on('message', (message) => {
     .then(() => { logger.info('Successfully added the role') })
 })
 
+// Kick users who haven't introduced themselves within two weeks
+let intervalid = client.setInterval((servers) => {
+  const itMuslimServer = servers.get(IT_MUSLIM_SERVER_ID)
+
+  const restrictedMembers = itMuslimServer.members.filter((member) => {
+    return !member.roles.has(FULL_MEMBER_ROLE_ID)
+  })
+
+  // Today's date
+  let now = new Date()
+
+  restrictedMembers.forEach((member) => {
+    if (member.user.bot) {
+      return
+    }
+
+    // Сalculation of the time interval in which the user was inactive
+    let daysInactive = (now.getTime() - member.joinedTimestamp) / MILLISECONDS_IN_DAY
+    if (daysInactive > 2 && daysInactive < 10) {
```

Comment:
I would additionally save `hoursInactive` and do something like:
```suggestion
    if (hoursInactive >= 7 * HOURS_IN_DAY && hoursInactive < (7 * HOURS_IN_DAY + 1)) {
```
Try to explain, why I decided to make this condition?